CIP-0108? | Governance Metadata - Governance Actions (#632)
cardano-foundation/CIPs: CIP-0108/README.md

By only supplying an onchain hash of the off-chain we ensure correctness of data whilst minimizing the amount of data posted to the chain.

### For voters
When observing from the chain level, tooling can only see the content of the governance action and it's anchor.
These on-chain components do not give give any context to the motivation nor off-chain discussion of an governance action.
Although this information would likely be desired context for voters.
By providing rich contextual metadata we enable voters to make well informed decisions.

### For all participants
By standardizing off-chain metadata formats we facilitate interoperability for tooling which creates and/or renders metadata attached to governance actions.
This intern promotes a rich user experience between tooling.
This is good for all governance participants.

## Specification
Although there are seven types of governance action defined via CIP-1694, we focus this proposal on defining core properties which must be attached to all types.
We leave room for future standards to refine and specialize further to cater more specific for each type of governance action.

> ### Markdown Text Styling

Reviewer: Supporting markdown means that we can allow rich text formatting without much concern that there will be malicious code injected (explorers can explicitly strip all HTML tags for safety or allow only a subset of tags). Generally speaking two blank lines equals a carriage return in Markdown but we could also declare a list of minimum HTML tags that can be supported such as the `<br />` tag.

> ### Markdown Text Styling

Author: Lets do it!

> ### Markdown Text Styling

Reviewer 2: @Ryun1 @Crypto2099 yes & by **not** including minimal HTML tags like `<i>` and `<b>` on the list we can promote standardisation of document format like Markdown's corresponding`*` (or `_`) and `**`.

Voters should be able to use this field to evaluate the applicability of the solution to the problem.

### `references`
References give the author ability to point to supporting research or related work.
These should be used by voters to verify the content of supporting research.
The inclusion of a hash allows for the supporting documentation to be cryptographically verified.

### Open Questions
- <s>Should fields be optional or compulsory?</s>
  - Title, abstract, motivation and rationale should be compulsory as they should be very important to the ability 
- <s>How much vocabulary can be extended to other onchain governance events?</s>
  - It is hard to predict how the scope of future standards before they have been developed.
- <s>How to integrate custom set of HTML tags? to allow formatting of longer text fields.</s>
  - Since CIP-100 does not intend to support rich text fields such an inclusion would not fit, so we have included such format here.

## Path to Active

### Acceptance Criteria
- [ ] This standard is supported by two different tooling providers used to submit governance actions to chain.

> - [ ] This standard is supported by two different chain indexing tools, used to read and render metadata.

Author: ```suggestion
- [ ] This standard is supported by two different chain indexing tools, used to read and render metadata.
  - DB-Sync via [db-sync-sancho-4.1.0](https://github.com/IntersectMBO/cardano-db-sync/releases/tag/sancho-4.1.0)
```

---
CIP: 108
Title: Governance Metadata - Governance Actions
Category: Metadata
Status: Proposed
Authors:
  - Ryan Williams <ryan.williams@intersectmbo.org>
Implementors: [ ]
Discussions:
  - https://github.com/cardano-foundation/CIPs/pull/632
  - https://github.com/cardano-foundation/CIPs/pull/748
  - https://github.com/cardano-foundation/CIPs/issues/757
Created: 2023-11-23
License: CC-BY-4.0
---

## Abstract
The Conway ledger era ushers in on-chain governance for Cardano via [CIP-1694 | A First Step Towards On-Chain Decentralized Governance](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1694/README.md), with the addition of many new on-chain governance artifacts.
Some of these artifacts support the linking off-chain metadata, as a way to provide context.

The [CIP-100 | Governance Metadata](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0100) standard provides a base framework for how all off-chain governance metadata should be formed and handled.
But this is intentionally limited in scope, so that it can be expanded upon by more specific subsequent CIPs.

> This proposal aims to provide a specification for off-chain metadata vocabulary can be used to give context to governance actions, CIP-100.

Author: ```suggestion
This proposal aims to provide a specification for off-chain metadata vocabulary that can be used to give context to governance actions.
```

grammar fix

- This SHOULD support markdown text styling.
- This SHOULD be used by the author to encapsulate all context around the problem that is being solved by the on-chain action.
- This SHOULD be used to outline the related stakeholders and use cases. 

#### `rationale`
- A freeform text field.
- This SHOULD support markdown text styling.
- This SHOULD be used by the author to discuss how the content of the governance action addresses the problem outlined within the `motivation`.
- This field SHOULD justify the changes being made to Cardano.
- i.e "by decreasing X parameter by Y we increase Ada earned by SPOs, thus incentivising more people to become SPOs, leading to a more diverse network"
- This SHOULD provide evidence of consensus within the community and discuss significant objections or concerns raised during the discussion.
- This SHOULD include discussion of alternative solutions and related topics/ governance actions.
- This SHOULD include any recommendations made by relevant organizations or committees.

#### `references`
- We extend CIP-100's references field.
- This SHOULD NOT support markdown text styling.
- To be an OPTIONAL set of objects, using the `@set` property.

> - Each object MUST have a `title` field to describe the reference, such as; "blog - Why we must continue to fund Catalyst".

Author: ```suggestion
- Each object MUST have a `label` field to describe the reference, such as; "blog - Why we must continue to fund Catalyst".
```

small fix

Author: Good spot @jdyczka
